src/index.js

```diff
@@ -84,7 +84,11 @@ function login(loginToken, requiredFields) {
                 try {
                     body = JSON.parse(body);
                     if (!body || !body.authenticationSuccess)
-                        err = new Error("Authentication failed");
+                        if (body.loginForm.includes('window.renderCaptcha()')) {
+                            return reject(new Error('USER_ACTION_NEEDED.CAPTCHA'));
```

Comment:
Why do you return the reject here? Setting `err` to `new Error('USER_ACTION_NEEDED.CAPTCHA')` should be enough?

Reply:
Also, this is weird: `body` is supposed to be a JSOn.parse result here

Reply:
If I set err to `new Error('USER_ACTION_NEEDED.CAPTCHA')` the code will reject with `LOGIN_FAILED`.

About `body`, it is JSON.parsed but body.loginForm still contains text which is actually html with javascript.

Reply:
Ok for both and sorry for the first one, indeed.